packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettingsSuspendClusterDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import { Engine } from '@linode/api-v4/lib/databases';
+import { useSnackbar } from 'notistack';
+import * as React from 'react';
+import { useHistory } from 'react-router-dom';
+import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
+import { Checkbox } from 'src/components/Checkbox';
+import { ConfirmationDialog } from 'src/components/ConfirmationDialog/ConfirmationDialog';
+import { Notice } from 'src/components/Notice/Notice';
+import { Typography } from 'src/components/Typography';
+import { useSuspendDatabaseMutation } from 'src/queries/databases/databases';
+import { getAPIErrorOrDefault } from 'src/utilities/errorUtils';
+
+export interface SuspendDialogProps {
+  databaseEngine: Engine;
+  databaseId: number;
+  databaseLabel: string;
+  onClose: () => void;
+  open: boolean;
+}
+
+export const DatabaseSettingsSuspendClusterDialog = (
+  props: SuspendDialogProps
+) => {
+  const { databaseEngine, databaseId, databaseLabel, onClose, open } = props;
+  const { enqueueSnackbar } = useSnackbar();
+  const {
+    mutateAsync: suspendDatabase,
+    isPending,
+    error,
+    reset,
+  } = useSuspendDatabaseMutation(databaseEngine, databaseId);
+
+  const defaultError = 'There was an error suspending this Database Cluster.';
+  const [hasConfirmed, setHasConfirmed] = React.useState(false);
+  const { push } = useHistory();
+
+  const onSuspendCluster = () => {
+    suspendDatabase()
+      .then(() => {
+        enqueueSnackbar('Database Cluster suspended successfully.', {
+          variant: 'success',
+        });
+        onClose();
+        setHasConfirmed(false);
+        push('/databases');
+      })
+      .catch(() => {
+        setHasConfirmed(false);
+      });
+  };
```

Comment:
nitpicky:  Can we use async/await to improve readability. try/catch handles success and error notifications, while finally ensures cleanup ?
```suggestion
const onSuspendCluster = async () => {
  try {
    await suspendDatabase();
    enqueueSnackbar('Database Cluster suspended successfully.', { variant: 'success' });
    onClose();
    push('/databases');
  } catch (error) {
    enqueueSnackbar('Failed to suspend Database Cluster. Please try again.', { variant: 'error' });
  } finally {
    setHasConfirmed(false);
  }
};
```

Reply:
I made this change in branch that we merged in.  (Sam is on vacation.)